### PR TITLE
SE - NonNullableValueTypes: SymbolicValue "True/False" also means "NotNull"

### PR DESCRIPTION
**Pull request by martin-strecker-sonarsource**
Part of #6105

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/SymbolicValue.cs [hunk 1]
```diff
@@ -28,8 +28,8 @@ public sealed record SymbolicValue
         public static readonly SymbolicValue This = new SymbolicValue().WithConstraint(ObjectConstraint.NotNull);
         public static readonly SymbolicValue Null = new SymbolicValue().WithConstraint(ObjectConstraint.Null);
         public static readonly SymbolicValue NotNull = new SymbolicValue().WithConstraint(ObjectConstraint.NotNull);
-        public static readonly SymbolicValue True = new SymbolicValue().WithConstraint(BoolConstraint.True);
-        public static readonly SymbolicValue False = new SymbolicValue().WithConstraint(BoolConstraint.False);
+        public static readonly SymbolicValue True = NotNull.WithConstraint(BoolConstraint.True);
+        public static readonly SymbolicValue False = NotNull.WithConstraint(BoolConstraint.False);
```
Comment (pavel-mikula-sonarsource) [inline analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/SymbolicValue.cs:32]: Oh, nice! 😆

File: analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Nullable.cs [hunk 2]
```diff
@@ -77,13 +77,12 @@ public void Nullable_HasValue_ReadsBoolConstraintFromObjectConstraint()
             Tag("SymbolAfterNull", arg);
             """;
         var validator = SETestContext.CreateCS(code, ", bool? arg").Validator;
-        validator.ValidateTag("HasValueUnknown", x => x.HasConstraint<BoolConstraint>().Should().BeFalse());
-        validator.ValidateTag("SymbolUnknown", x => x.Should().BeNull());
-        validator.ValidateTag("HasValueAfterTrue", x => x.HasConstraint<BoolConstraint>().Should().BeFalse());   // ToDo: Should be x.HasConstraint(BoolConstraint.True).Should().BeTrue()); after we build NotNull for bool literal.
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Nullable.cs:82]: 🎉

File: analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Nullable.cs [hunk 3]
```diff
@@ -77,13 +77,12 @@ public void Nullable_HasValue_ReadsBoolConstraintFromObjectConstraint()
...
-        validator.ValidateTag("SymbolAfterTrue", x => x.HasConstraint(BoolConstraint.True).Should().BeTrue());
-        validator.ValidateTag("SymbolAfterTrue", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeFalse());    // ToDo: Should be BeTrue() after we build NotNull for bool literal
-        validator.ValidateTag("HasValueAfterNull", x => x.HasConstraint(BoolConstraint.False).Should().BeTrue());
-        validator.ValidateTag("SymbolAfterNull", x => x.HasConstraint(ObjectConstraint.Null).Should().BeTrue());
+        validator.ValidateTag("HasValueUnknown", x => x.Should().HaveOnlyConstraint(ObjectConstraint.NotNull));
+        validator.ValidateTag("SymbolUnknown", x => x.Should().HaveNoConstraints());
+        validator.ValidateTag("HasValueAfterTrue", x => x.Should().HaveOnlyConstraints(ObjectConstraint.NotNull, BoolConstraint.True));
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Nullable.cs:82]: There's a rebase conflict. I think I've rewritten this UT to use `int?` instead, to avoid this pitfall. So you can probably rebase and revert these changes. Or re-add this UT as a 2nd test for this.

I don't have a preference.

**Review by pavel-mikula-sonarsource — APPROVED** [review]
LGTM

File: analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Nullable.cs (hunk 3 above)
Comment (martin-strecker-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Nullable.cs]: I added the `bool?` test (out of curiosity if everything still works).

**Comment by martin-strecker-sonarsource** [comment]
@pavel-mikula-sonarsource can you please merge (it is still blocked due to the ITs issue).
